src/main/java/io/vertx/httpproxy/impl/ProxyRequestImpl.java

```diff
@@ -193,4 +185,14 @@ void send(HttpClientRequest inboundRequest, Handler<AsyncResult<ProxyResponse>>
     this.inboundRequest = inboundRequest;
     sendRequest(completionHandler);
   }
+
+  void websocket(WebSocket inboundWebSocket) {
+    Future<ServerWebSocket> fut = outboundRequest.toWebSocket();
+    fut.onSuccess(ws -> {
```

Comment:
we should handle and test the failure case

Reply:
The relevant code has been improved, but I am not sure how the unit test is written.